Add/client side media uploads

adamsilverstein opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Follow up to https://github.com/WordPress/gutenberg/pull/64278, this PR aims to provide only the minimal changes required to move media processing (eg. sub-sized image generation when users upload images) _to the client from the server_.

See https://github.com/WordPress/gutenberg/issues/61447 for full context.


### Follow up tasks to align with core
All of these features are already handled in the media-experiments plugins and only need to be ported/polished for Gutenberg.

- [ ] Support core output mapping (from the `image_editor_output_format` filter)
- [ ] Support for all types including video, PDF, WebP, HEIF and AVIF.
- [ ] Handle image auto-rotation based on exif data (similar to core)
- [ ] Scale down images according to big image size threshold
- [ ] Ensure correct handling of PNGs: Indexed, true color, transparencies, grayscale.
- [ ] Support for UltraHDR

Also, this will work out of the box and be a great addition:
- [X] Use MozJPEG for JPEG image saves - this will provide near WebP like sizes with 100% backwards compatibility

## Why?
https://github.com/WordPress/gutenberg/pull/64278 was no longer building correctly and has many conflicting files in trunk. Reviewers mentioned several times reducing the scope of the PR to to reduce review effort.

## How?
This PR starts with [#64278](https://github.com/WordPress/gutenberg/pull/64278), brings it current with trunk and removes changes unrelated to initial task: "Client-side sub-size image generation". Files changed are reduced to 116 to 52, although some additional changes may be required.  It removes the `@wordpress/mime` package as requested and doesn't make changes to the image block.

## Testing Instructions
1. Ensure experiments->client side media is enabled
2. Upload an image in the block editor
3. Monitor the network panel for activity as each image is updated. Check the files saved to the server and compare them to the files generated when the feature is not enabled.


 ## Todo
- [ ] Switch to loading wasm-vips locally, instead of from the CDN. @swissspidy has already worked out a potential solution in https://github.com/swissspidy/media-experiments/pull/753.
- [ ] Switch from the deprecated "@shopify/web-worker" package (previously https://github.com/WordPress/gutenberg/issues/69254)
- [x] Fix failing tests

swissspidy committed:
First pass at porting over code

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

adamsilverstein committed:
remove wordpress/mime package

adamsilverstein committed:
types fix

adamsilverstein committed:
Fix build after merge; remove timings (sans docs)

adamsilverstein committed:
cleanup

adamsilverstein committed:
remove unused styles

adamsilverstein committed:
remove duplcate change

adamsilverstein committed:
remove duplcate change

adamsilverstein committed:
remove unneeded change

adamsilverstein committed:
cleanup

adamsilverstein committed:
Cleanup

adamsilverstein committed:
cleanup

adamsilverstein commented on packages/upload-media/package.json:
```diff
@@ -28,6 +28,7 @@
 	"sideEffects": false,
 	"dependencies": {
 		"@babel/runtime": "7.25.7",
+		"@shopify/web-worker": "^6.3.0",
```
Looks like this package is no longer supported and should ideally be removed: https://www.npmjs.com/package/@shopify/web-worker

adamsilverstein replied:
working on this in https://github.com/adamsilverstein/gutenberg/pull/25

adamsilverstein committed:
Add note about @shopify/web-worker deprecation

adamsilverstein committed:
Doc updates

adamsilverstein committed:
update block editor actions and settings

adamsilverstein committed:
remove duplicate files

adamsilverstein committed:
vips load cleanup

adamsilverstein committed:
more vips cleanup

adamsilverstein committed:
mime data fixes

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

adamsilverstein committed:
test and type fixes

adamsilverstein committed:
test fixes

also:
- complete mime removal, inlining code where it is used or as a local helper function
- complete removal of approval requirement feature
- add back pause action and resume action their absence was causing test failures. They seem useful to include although I don’t think they are strictly required for this PR
- ensure testing uses images not videos for urls (is video technically supported already?)
- make expects declarations more narrow and explicit in `optimizeExistingItem` test
- complete timings removal (was still in reducer test)

adamsilverstein commented:
> Switch from the deprecated "@shopify/web-worker" package (previously https://github.com/WordPress/gutenberg/issues/69254)

I started a try on this in https://github.com/adamsilverstein/gutenberg/pull/25

adamsilverstein committed:
complete mime removal

adamsilverstein committed:
linting

adamsilverstein committed:
more linting/test fixes

adamsilverstein committed:
more linting/test fixes, remove unused files

adamsilverstein committed:
remove resize images tests for now, readd once wasm-vips is loaded locally

adamsilverstein committed:
remove sizing tests for now, __experimentalUpdateSettings tests should expect dispatch

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

adamsilverstein committed:
restore missing setInsertionPoint action

adamsilverstein committed:
restore missing code in private actions

adamsilverstein committed:
Fix __unstableIsPreviewMode handling

adamsilverstein committed:
restore package-lock from trunk

adamsilverstein committed:
move settings to existing `gutenberg_get_block_editor_settings` callback

adamsilverstein committed:
re-add @shopify/web-worker

adamsilverstein committed:
Revert "move settings to existing `gutenberg_get_block_editor_settings` callback"

This reverts commit 6bb1e8890d2069f6f1b272b327b249d3c24bb737.

adamsilverstein committed:
remove unused wasm-vips patch

adamsilverstein committed:
update wasm-vips version

adamsilverstein committed:
Add asyncWebAssembly: true to webpack.config.js

adamsilverstein committed:
add module mapping for wasm-vips

adamsilverstein committed:
update build

adamsilverstein committed:
yoda conditional

adamsilverstein committed:
revert excludes in webpack

adamsilverstein committed:
revert config and isProduction changes

adamsilverstein committed:
revert additional webpack changes

adamsilverstein commented:
@swissspidy Took a bit of work, tests are passing!

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

adamsilverstein committed:
remove grantApproval and ApproveUploadAction

adamsilverstein committed:
remove rejectApproval

swissspidy commented on lib/experimental/media/load.php:
```diff
@@ -360,3 +360,28 @@ static function (): void {
 
 add_action( 'wp_enqueue_media', 'gutenberg_override_media_templates' );
+
+/**
+ * Adds media-related settings to the block editor.
+ *
+ * @since X.X.X
+ *
+ * @param array $settings Existing block editor settings.
+ *
+ * @return array New block editor settings.
+ */
+function gutenberg_block_editor_settings_media_processing( $settings ) {
+	if ( ! gutenberg_is_experiment_enabled( 'gutenberg-media-processing' ) ) {
+		return $settings;
+	}
+
+	/** This filter is documented in wp-admin/includes/images.php */
+	$image_size_threshold = (int) apply_filters( 'big_image_size_threshold', 2560, array( 0, 0 ), '', 0 ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+
+	$settings['__experimentalAvailableImageSizes']   = gutenberg_get_all_image_sizes();
+	$settings['__experimentalBigImageSizeThreshold'] = $image_size_threshold;
+
+	return $settings;
+}
+
+add_filter( 'block_editor_settings_all', 'gutenberg_block_editor_settings_media_processing' );
```
The Gutenberg team originally suggested moving these to the REST API instead of providing these as block editor settings.

See https://github.com/swissspidy/media-experiments/issues/561, https://github.com/swissspidy/media-experiments/pull/621 and https://github.com/swissspidy/media-experiments/blob/ec464e8a1a1e450a5f4610458a2d66ad3a141b31/inc/functions.php#L666-L721

The settings can then be retrieved via `getEntityRecord( 'root', '__unstableBase' )`

adamsilverstein replied:
I tried working on this a bit and ran into issues where importing the core datastore in the upload-media component seemed to cause a circular reference (PHP ran out of memory trying to load the page).

Any tips on where/how to add this or a link to the original suggestion?

swissspidy replied:
The fetching of the settings needs to happen somewhere in the editor package, which can then call updateSettings() from the upload-media package.

in the plugin I did it like this: https://github.com/swissspidy/media-experiments/blob/ec464e8a1a1e450a5f4610458a2d66ad3a141b31/packages/editor/src/init/index.ts#L78

But in GB we can avoid doing a subscribe() call I think.

adamsilverstein replied:
thanks, that confirms what I was thinking, I will give that a try.

adamsilverstein replied:
Changed in daa6474d3a0e3d310b7525bdb9e7325a35a3ace6

swissspidy commented on packages/preferences/package.json:
```diff
@@ -28,6 +28,7 @@
 	"module": "build-module/index.js",
 	"react-native": "src/index",
 	"wpScript": true,
+	"types": "build-types",
```
Can we split out the changes to this package into a separate PR?

I originally attempted this in #64799, it could probably be resurrected.

swissspidy commented on packages/upload-media/package.json:
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@wordpress/upload-media",
-	"version": "0.11.0",
+	"version": "1.0.0-prerelease",
```
need to undo this one

adamsilverstein replied:
done in 2bec17cad72af4d9a85c6b35ea6559b61ab4c379

swissspidy commented on packages/vips/package.json:
```diff
@@ -28,7 +28,9 @@
 	"types": "build-types",
 	"dependencies": {
-		"wasm-vips": "^0.0.10"
+		"@types/node-fetch": "2.6.12",
+		"node-fetch": "2.7.0",
```
Why node-fetch? shouldn't this be in a browser?

adamsilverstein replied:
this was to get tests passing that used Vips which is fetched from the CDN currently (`await ( await fetch( `${ VIPS_CDN_URL }/vips.js` ) ).blob()`). Once we switch to loading locally/embedding, we can remove this.

swissspidy replied:
But don't we have `window.fetch` also in tests?

adamsilverstein replied:
> But don't we have window.fetch also in tests?

I'm not sure, let me see if any other tests use fetch or try that.

adamsilverstein replied:
that seemed to work, I changed it in https://github.com/WordPress/gutenberg/pull/70623/commits/9de8237059982dfa07f19f803f89342e8b042734

swissspidy commented on packages/upload-media/src/store/workers/canvas.ts:
```diff
@@ -0,0 +1,195 @@
+/**
```
Maybe we don't actually want to preserve this. It's like a fallback for when vips isn't available. But canvas is really poor for optimizing images.

adamsilverstein replied:
Ok, I will remove. I prefer to keep this simple anyway!

adamsilverstein replied:
removed in 30b0be7172844ed3b3e5e322871cb6e44ecebfbd

swissspidy commented on packages/upload-media/src/store/actions.ts:
```diff
@@ -132,15 +158,156 @@ export function addItems( {
+ * @param [$0.additionalData]    Additional data to include in the request.
+ * @param [$0.generatedPosterId] Attachment ID of the generated poster image, if it exists.
+ */
+export function optimizeExistingItem( {
```
Out of scope for this PR?

adamsilverstein replied:
will remove, thanks

adamsilverstein replied:
removed in 55b2deab23f4c666554d3fea0023f69ef3256db1

swissspidy commented on packages/upload-media/src/store/actions.ts:
```diff
@@ -132,15 +158,156 @@ export function addItems( {
+ * @param [$0.onError]        Function called when an error happens.
+ * @param [$0.additionalData] Additional data to include in the request.
+ */
+export function addItemFromUrl( {
```
Out of scope for this PR?

adamsilverstein replied:
right, will remove these parts

adamsilverstein replied:
removed in e6cd2bcc11c99c74b285684961b13060e1155c6c

adamsilverstein committed:
Remove packages/upload-media/src/store/workers/canvas.ts

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

adamsilverstein committed:
Remove `addItemFromUrl`

adamsilverstein committed:
Remove

adamsilverstein committed:
Revert version change in package.json

adamsilverstein committed:
Try windows.fetch

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

t-hamano committed:
Improve plugin release checklist and documentation (#70808)

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

mukeshpanchal27 committed:
Fix: Avoid duplicate `get_the_title()` calls in `render_block_core_post_featured_image()` (#70813)

Co-authored-by: mukeshpanchal27 <mukesh27@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>

chihsuan committed:
DataViews: Add `date` field type (#70657)

Co-authored-by: chihsuan <chihsuan@git.wordpress.org>
Co-authored-by: oandregal <oandregal@git.wordpress.org>

Infinite-Null committed:
Docs: Update RichTextToolbarButton link in docs (#70804)

Co-authored-by: Infinite-Null <ankitkumarshah@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>
Co-authored-by: atachibana <atachibana@git.wordpress.org>

oandregal committed:
DataViews: improve story dataset (#70817)

Co-authored-by: oandregal <oandregal@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

Mamaduka committed:
Navigation Link: Fix URL deletion in Inspector Controls (#70749)

* Navigation Link: Fix URL deletion in Inspector Controls
* Fix unit tests
* Defer should sever entity link logic

Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>
Co-authored-by: getdave <get_dave@git.wordpress.org>
Co-authored-by: scruffian <scruffian@git.wordpress.org>
Co-authored-by: Robertght <robertghetau@git.wordpress.org>

yogeshbhutkar committed:
[Block] Cover: Fix resize handler gets stuck on undo (#70820)


Co-authored-by: yogeshbhutkar <yogeshbhutkar@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

oandregal committed:
DataViews: Add groupByField support to grid layout (#70752)

Co-authored-by: oandregal <oandregal@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

adamsilverstein committed:
fix loading of wasm-vips

swissspidy commented on .github/ISSUE_TEMPLATE/New_release.md:
Unrelated change to this docs file and others.

adamsilverstein replied:
let me regenerate docs to see if these go away.

adamsilverstein replied:
updating from trunk resolved these

swissspidy commented on packages/block-library/src/navigation-link/edit.js:
What's with the many changes here?

adamsilverstein replied:
possibly failed merge resolution from trunk, will revert.

adamsilverstein replied:
updated from trunk

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

adamsilverstein committed:
revert tools/webpack/shared.js change

adamsilverstein committed:
Remove experimental media processing settings from block editor configuration, grab from REST API instead

t-hamano committed:
Fields: Add label to Password field (#70865)


Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

kushagra-goyal-14 committed:
TypeScript: Migrate jest-console package to TS (#70538)

Co-authored-by: kushagra-goyal-14 <kush123@git.wordpress.org>
Co-authored-by: manzoorwanijk <manzoorwanijk@git.wordpress.org>

kushagra-goyal-14 committed:
TypeScript: migrate priority-queue package to TS (#70469)

Co-authored-by: kushagra-goyal-14 <kush123@git.wordpress.org>
Co-authored-by: manzoorwanijk <manzoorwanijk@git.wordpress.org>

Mamaduka committed:
Editor: Fix 'isEditedPostAutosaveable' selector return value (#70888)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: swissspidy <swissspidy@git.wordpress.org>

shimotmk committed:
Zoom Out: Delete unused property (#70893)


Co-authored-by: shimotmk <shimotomoki@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

swissspidy commented on packages/block-editor/src/store/defaults.js:
Looks like these changes can be reverted now that the settings are loaded from the REST API.

adamsilverstein replied:
removed

swissspidy commented on packages/block-editor/README.md:
Looks like these changes can be reverted now that the settings are loaded from the REST API.

swissspidy commented on packages/editor/src/utils/media-upload/index.ts:
```diff
@@ -74,41 +90,46 @@ export default function mediaUpload( {
+				null
+			);
+			clearSaveLock();
+			onSuccess?.( entityFiles as Attachment[] );
```
Why not pass `onSuccess` separately to `uploadMedia` as is done today?

swissspidy commented on packages/upload-media/src/store/actions.ts:
```diff
@@ -59,6 +62,23 @@ type ThunkArgs = {
 };
 
+/**
+ * Returns an action object that pauses all processing in the queue.
```
Looks like I didn't use the correct docblock here in the media-experiments plugin.

swissspidy commented on packages/upload-media/src/store/actions.ts:
```diff
@@ -89,8 +111,9 @@ export function addItems( {
 	onBatchSuccess,
 	additionalData,
 	allowedTypes,
+	maxUploadFileSize,
```
Hmm why is this now an argument instead of being grabbed from the settings?

----

Side note: not sure why `addItems` here is so different from the one in media-experiments 🤔

swissspidy commented on packages/upload-media/src/store/actions.ts:
```diff
@@ -155,16 +173,22 @@ export function cancelItem( id: QueueItemId, error: Error, silent = false ) {
+
 		item.abortController?.abort();
 
-		if ( ! silent ) {
```
Why remove this logic?

swissspidy commented on packages/upload-media/src/store/types.ts:
```diff
@@ -4,7 +4,7 @@ export type QueueStatus = 'active' | 'paused';
 export type BatchId = string;
 
-export interface QueueItem {
+export type QueueItem = {
```
I think we'll want to keep `interface` here.

swissspidy commented on packages/upload-media/src/store/types.ts:
```diff
@@ -120,17 +125,30 @@ interface UploadMediaArgs {
 }
 
+export type Settings = {
+	mediaUpload: ( args: UploadMediaArgs ) => void;
+	mediaSideload: ( args: SideloadMediaArgs ) => void;
+	imageSizes: Record< string, ImageSizeCrop >;
+	imageSizeThreshold: number;
+};
+
 // Must match the Attachment type from the media-utils package.
-export interface Attachment {
+export type Attachment = {
```
I think we'll want to keep `interface` here too

swissspidy commented on packages/vips/src/index.ts:
```diff
@@ -240,7 +216,7 @@ export async function resizeImage(
 	};
 
-	let image = vips.Image.newFromBuffer( buffer, strOptions, loadOptions );
+	let image = vips.Image.newFromBuffer( buffer ); //  buffer, strOptions, loadOptions );
```
?

dsas committed:
Update dataviews changelog for #70785 (#70885)

It was missed when merging #70785 originally.

Co-authored-by: dsas <dsas@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

Mamaduka committed:
Categories: Disable links in the editor (#70896)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: yogeshbhutkar <yogeshbhutkar@git.wordpress.org>
Co-authored-by: talldan <talldanwp@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: ajaxStardust <ajaxstardust@git.wordpress.org>
Co-authored-by: sabernhardt <sabernhardt@git.wordpress.org>

t-hamano committed:
Change header height from 60px to 64px (#60729)

* Change header height from 60px to 64px

* Improve comments for vertical block mover

* Add comment for Modal style

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: jameskoster <jameskoster@git.wordpress.org>
Co-authored-by: richtabor <richtabor@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>

jameskoster committed:
Add icon: Dashboard (#70300)


Co-authored-by: jameskoster <jameskoster@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>

KHarika01 committed:
Fix: Add hyphen to “hands-on” in Getting Started section (#70906)

Small grammar improvement in the README.md file.

Unlinked contributors: KHarika01.

Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

arthur791004 committed:
DataViews: Align media styles in table view with list view for consistency (#70671)

Co-authored-by: arthur791004 <arthur791004@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>
Co-authored-by: matt-west <mattwest@git.wordpress.org>
Co-authored-by: jameskoster <jameskoster@git.wordpress.org>

BugReportOnWeb committed:
File Block: Fix PDF preview on URLs containing query parameters (#70915)

* fix: detect PDF urls with query strings in file block

* fix: use `getFilename` to fetch filename from media URL

Unlinked contributors: MichaelCastelbuono.

Co-authored-by: BugReportOnWeb <devasheeshkaul@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>

t-hamano committed:
RichText: Support more colors for format boundary colors (#70918)

Unlinked contributors: vaibhavshinde8478, ktmn.

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

shail-mehta committed:
Updated Small Typo in block-styles.md file (#70927)

Co-authored-by: shail-mehta <shailu25@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>

westonruter committed:
Fix layout shift caused by `video` tag in Video block lacking `width` and `height` (#70293)

* Prevent layout shift in Video block when width/height metadata is available

* Add todo for capturing dimensions in JS to persist in block attributes

* Only set auto:height on videos with width and height set

* Use height:auto regardless of whether the width/height attributes were supplied

* Add explanation for why inline style is added

* Fix syntax of attr()

* Update since tag

Co-authored-by: George Mamadashvili <georgemamadashvili@gmail.com>

* Fix other since tag

---------

Unlinked contributors: porg.

Co-authored-by: westonruter <westonruter@git.wordpress.org>
Co-authored-by: adamsilverstein <adamsilverstein@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: priethor <priethor@git.wordpress.org>

westonruter committed:
Add `@wordpress/env` as an optional peer dependency of `@wordpress/scripts` (#70636)

Unlinked contributors: renatonascalves.

Co-authored-by: westonruter <westonruter@git.wordpress.org>
Co-authored-by: swissspidy <swissspidy@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: peterwilsoncc <peterwilsoncc@git.wordpress.org>
Co-authored-by: pmbaldha <pmbaldha@git.wordpress.org>

t-hamano committed:
TextareaControl: Add missing component CSS classname (#70930)

* TextareaControl: Add missing component CSS classname

* Add changelog

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

yogeshbhutkar committed:
Page List: Fix Notice component background gets incorrectly inherited (#70932)

Co-authored-by: yogeshbhutkar <yogeshbhutkar@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>

t-hamano committed:
PaletteEdit: Fill available space with input field (#70935)

* PaletteEdit: Fill available space with input field

* Update changelog

Mamaduka committed:
Block Editor: Fix link interception in the iframed editor (#70931)

* Block Editor: Fix link interception in the iframed editor
* Adjust comments

Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: mcsf <mcsf@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>

mirka committed:
TextControl: Improve theming support (#70910)

* TextControl: Improve theming support

* Add changelog

dependabot[bot] committed:
Bump shivammathur/setup-php (#70944)

Bumps the github-actions group with 1 update in the /.github/workflows directory: [shivammathur/setup-php](https://github.com/shivammathur/setup-php).


Updates `shivammathur/setup-php` from 2.34.1 to 2.35.1
- [Release notes](https://github.com/shivammathur/setup-php/releases)
- [Commits](https://github.com/shivammathur/setup-php/compare/0f7f1d08e3e32076e51cae65eb0b0c871405b16e...2282b6a082fc605c8320908a4cca3a5d1ca6c6fe)

---
updated-dependencies:
- dependency-name: shivammathur/setup-php
  dependency-version: 2.35.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: desrosj <desrosj@git.wordpress.org>

dependabot[bot] committed:
Chore(deps): bump http-proxy-middleware in /platform-docs (#69946)

Bumps [http-proxy-middleware](https://github.com/chimurai/http-proxy-middleware) from 2.0.7 to 2.0.9.
- [Release notes](https://github.com/chimurai/http-proxy-middleware/releases)
- [Changelog](https://github.com/chimurai/http-proxy-middleware/blob/v2.0.9/CHANGELOG.md)
- [Commits](https://github.com/chimurai/http-proxy-middleware/compare/v2.0.7...v2.0.9)

---
updated-dependencies:
- dependency-name: http-proxy-middleware
  dependency-version: 2.0.9
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: desrosj <desrosj@git.wordpress.org>

dependabot[bot] committed:
Bump form-data from 4.0.0 to 4.0.4 (#70829)

Bumps [form-data](https://github.com/form-data/form-data) from 4.0.0 to 4.0.4.
- [Release notes](https://github.com/form-data/form-data/releases)
- [Changelog](https://github.com/form-data/form-data/blob/master/CHANGELOG.md)
- [Commits](https://github.com/form-data/form-data/compare/v4.0.0...v4.0.4)

---
updated-dependencies:
- dependency-name: form-data
  dependency-version: 4.0.4
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: desrosj <desrosj@git.wordpress.org>

dependabot[bot] committed:
Bump on-headers and compression in /platform-docs (#70892)

Bumps [on-headers](https://github.com/jshttp/on-headers) and [compression](https://github.com/expressjs/compression). These dependencies needed to be updated together.

Updates `on-headers` from 1.0.2 to 1.1.0
- [Release notes](https://github.com/jshttp/on-headers/releases)
- [Changelog](https://github.com/jshttp/on-headers/blob/master/HISTORY.md)
- [Commits](https://github.com/jshttp/on-headers/compare/v1.0.2...v1.1.0)

Updates `compression` from 1.7.4 to 1.8.1
- [Release notes](https://github.com/expressjs/compression/releases)
- [Changelog](https://github.com/expressjs/compression/blob/master/HISTORY.md)
- [Commits](https://github.com/expressjs/compression/compare/1.7.4...v1.8.1)

---
updated-dependencies:
- dependency-name: on-headers
  dependency-version: 1.1.0
  dependency-type: indirect
- dependency-name: compression
  dependency-version: 1.8.1
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: desrosj <desrosj@git.wordpress.org>

dependabot[bot] committed:
Bump brace-expansion from 1.1.11 to 1.1.12 in /platform-docs (#70465)

Bumps [brace-expansion](https://github.com/juliangruber/brace-expansion) from 1.1.11 to 1.1.12.
- [Release notes](https://github.com/juliangruber/brace-expansion/releases)
- [Commits](https://github.com/juliangruber/brace-expansion/compare/1.1.11...v1.1.12)

---
updated-dependencies:
- dependency-name: brace-expansion
  dependency-version: 1.1.12
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: desrosj <desrosj@git.wordpress.org>

Mamaduka committed:
Block API: Add special case for metadata attribute in 'isUnmodifiedBlock' (#70903)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: cbravobernal <cbravobernal@git.wordpress.org>

heavyweight committed:
Core Data: Add getTitle for user entity (#70941)


Co-authored-by: heavyweight <kpapazov@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

Mamaduka committed:
Documentation: Improve note for 'blockType.save' method (#70942)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>

Mamaduka committed:
ServerSideRender: Introduce a new 'useServerSideRender' hook (#70543)


Unlinked contributors: amir2mi.

Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: jsnajdr <jsnajdr@git.wordpress.org>
Co-authored-by: Soean <soean@git.wordpress.org>
Co-authored-by: gziolo <gziolo@git.wordpress.org>
Co-authored-by: oxyc <scholdstrom@git.wordpress.org>

im3dabasia committed:
Components: add `placement` prop to replace `position` in `RangeControl` (#70326)


Co-authored-by: im3dabasia <im3dabasia1@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

talldan committed:
Dataviews: Add ctrl/cmd + click multiselection support to table layout when clicking rows (#70891)

* Add non-consecutive multi-selection to table view

* Use correct modifier key in grid view for each OS

* Add keycodes dependency

* Add tests for single and multiselection for table and grid

* Add keycodes to tsconfig

----

Co-authored-by: talldan <talldanwp@git.wordpress.org>
Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>
Co-authored-by: tellthemachines <isabel_brison@git.wordpress.org>

talldan committed:
Add changelog entry (#70953)

Co-authored-by: talldan <talldanwp@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>

Mamaduka committed:
ServerSideRender: Add missing changelog entry (#70952)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>

yogeshbhutkar committed:
Video/Cover Block: Enhance UI for Poster Image Preview (#70939)


Co-authored-by: yogeshbhutkar <yogeshbhutkar@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>

ockham committed:
Date block: Allow connecting to Block Bindings (#70585)

Make the Date block accept a manually set date, i.e. independent of the current post's publish or last modified date. Retain the latter two options by introducing a new Block Bindings source called `core/post-data` that exposes those two dates.

Update existing block variations of the Date block to continue to allow easy insertion of the "Post Date" and "Last Modified Date" variations of the block.

Include code for back-compat in the block's PHP, and a block deprecation to update existing Date blocks to use the new Block Bindings-based approach.

---------

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

ramonjd committed:
Core data: add a private media "edit" action to ensure cache invalidation (#70814)

This commit adds a new private action `editMediaEntity` to Core Data and associated unit tests. Because it needs to be called in block-editor hooks, the action is accessible in block editor settings via a private key. The primary motivation is to provide a core data integration for the `WP_REST_Attachments_Controller` REST API's `wp/v2/media/{ id }/edit` route. 


Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>
Co-authored-by: talldan <talldanwp@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>

Mamaduka committed:
Social Icon: Retire the Skype social service (#70769)

* Social Icon: Retire the Skype social service
* Add an inline comment

Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>
Co-authored-by: talldan <talldanwp@git.wordpress.org>
Co-authored-by: chad1008 <shireling@git.wordpress.org>

Sahil1617 committed:
Deprecate data-wp-ignore directive with warning message (#70945)

* Deprecate data-wp-ignore directive with warning message

* Update packages/interactivity/src/directives.tsx

Co-authored-by: David Arenas <darensr@gmail.com>

* Add deprecation note for data-wp-ignore directive in CHANGELOG

---------

Co-authored-by: Sahil1617 <sahiljadhav1617@git.wordpress.org>
Co-authored-by: DAreRodz <darerodz@git.wordpress.org>
Co-authored-by: USERSATOSHI <tusharbharti@git.wordpress.org>

jsnajdr committed:
Explain why we need to intercept hash links in iframe editor (#70960)

Co-authored-by: jsnajdr <jsnajdr@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

t-hamano committed:
Social Link Block: Add context to service names (#70413)

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: alexclassroom <alexclassroom@git.wordpress.org>

Mamaduka committed:
Paragraph Block: Fix 'Drop cap' control reset value (#70959)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: fabiankaegy <fabiankaegy@git.wordpress.org>

chihsuan committed:
Docs: Clarify filtering behavior in DataViews configuration in README.md (#70958)

Co-authored-by: chihsuan <chihsuan@git.wordpress.org>

gutenbergplugin committed:
Bump plugin version to 21.3.0-rc.2

oandregal committed:
DataViews: clear user-input filters properly (#70956)

Co-authored-by: oandregal <oandregal@git.wordpress.org>
Co-authored-by: chihsuan <chihsuan@git.wordpress.org>

gutenbergplugin committed:
Update Changelog for 21.3.0-rc.2

himanshupathak95 committed:
Color Picker: Improve color picker slider focus styles (#70146)

* Color Picker: Improve color picker slider focus styles

* Add respective changelog entry

oandregal committed:
DataForm: bootstrap validation (required and typechecks) (#70901)

Co-authored-by: oandregal <oandregal@git.wordpress.org>
Co-authored-by: mcsf <mcsf@git.wordpress.org>
Co-authored-by: gigitux <gigitux@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>

dependabot[bot] committed:
Bump pbkdf2 from 3.1.2 to 3.1.3 (#70502)

Bumps [pbkdf2](https://github.com/crypto-browserify/pbkdf2) from 3.1.2 to 3.1.3.
- [Changelog](https://github.com/browserify/pbkdf2/blob/master/CHANGELOG.md)
- [Commits](https://github.com/crypto-browserify/pbkdf2/compare/v3.1.2...v3.1.3)

---
updated-dependencies:
- dependency-name: pbkdf2
  dependency-version: 3.1.3
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: desrosj <desrosj@git.wordpress.org>

ramonjd committed:
useSaveImage: handle missing editMediaEntity gracefully (#70967)

Fix useSaveImage to handle missing editMediaEntity gracefully
- Updated the retrieval of editMediaEntity to use optional chaining.
- Added error handling to notify users when no edit media entity action is found during image editing.


Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

chihsuan committed:
DataViews: Implement Edit control for date field (#70836)

Co-authored-by: chihsuan <chihsuan@git.wordpress.org>
Co-authored-by: oandregal <oandregal@git.wordpress.org>

gutenbergplugin committed:
Bump plugin version to 21.3.0

gutenbergplugin committed:
Update Changelog for 21.3.0

t-hamano committed:
List View: Fix focus outline style (#69201)

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: andrewserong <andrewserong@git.wordpress.org>
Co-authored-by: afercia <afercia@git.wordpress.org>

Mamaduka committed:
ServerSideRender: Allow skipping block shadow support value (#70970)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: lipemat <mat-lipe@git.wordpress.org>

Mamaduka committed:
Block Editor: Fix single block selection when holding 'Shift' key (#70978)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: tyxla <tyxla@git.wordpress.org>

Mamaduka committed:
Social: Document guidelines and extensibility (#70961)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: bph <bph@git.wordpress.org>
Co-authored-by: priethor <priethor@git.wordpress.org>

westonruter committed:
Fix ability to preview Additional CSS changes in the Customizer when using a Block Theme (#70428)

* Fix ability to preview Custom CSS changes in the Customizer

* Add backport changelog

https://github.com/WordPress/wordpress-develop/pull/9000
https://github.com/WordPress/gutenberg/pull/70428

* Undo unnecessary update to deprecated function

* Ensure global styles are shown in Customizer preview and override Customizer styles in cascade

* Add Customizer details to the comment

* Update Customizer Custom CSS inside of global styles for live preview

* Make sure saved milestone comments are removed from stored CSS when previewing

* Add line breaks before/after Custom CSS for readability/debugging

Co-authored-by: Peter Wilson <peterwilsoncc@git.wordpress.org>

* Remove erroneous duplicated appending of global styles custom CSS

* Simplify adding inline script to customize-preview

Co-authored-by: t-hamano <wildworks@git.wordpress.org>

* Improve JS formatting

* Add commit reference

* Move custom CSS preview logic to 6.9 compat include

Co-authored-by: t-hamano <wildworks@git.wordpress.org>

---------

Co-authored-by: westonruter <westonruter@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Peter Wilson <peterwilsoncc@git.wordpress.org>

ockham committed:
Post Date block: Address feedback (#70982)

* Block bindings: Translate core/post-data source labels
* Use epoch to compare dates
* Add to backport changelog

jeryj committed:
Add shuffle style to write mode toolbar (#70946)

* Add shuffle style to content only toolbar
* Only show shuffle icon if write mode is enabled
* Limit shuffle section style block toolbar button to section blocks

tellthemachines committed:
Try using responsive images in dataview grid layouts (#70493)

* Try using responsive images in dataview grid layouts

* Move determination of sizes to dataviews

* Simplify grid logic and sizes

* use range of image widths for grid size picker

* Make picker discrete and improve breakpoints

* remove unused prop

* ref no worky again

* smol improvement

* Update to use an object instead of sizes string and fix grid size at max

* Adjust table max img width

* Change `mediaAppearance` to the more generic `config`

* Change `size` to `sizes`

* Explain magic number in comment.

dhruvikpatel18 committed:
Migrate undo-manager package to TS (#70757)

Co-authored-by: dhruvikpatel18 <dhruvik18@git.wordpress.org>
Co-authored-by: manzoorwanijk <manzoorwanijk@git.wordpress.org>

oandregal committed:
Add DataForm story about field visibility (#70996)

t-hamano committed:
Editor: Fix incorrect title when multiple post selected (#70997)

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>

vipul0425 committed:
iAPI Router: Ignores `<noscript>` elements while preparing DOM. (#70905)

* feat: Ignores <noscript> elements while preparing DOM.

* Docs: Update change logs with recent changes.

* feat: Adds e2e tests for noscript fix.

* docs: Added comment on code block.

* Remove nonexistent stylesheet from noscript tests

---------

Co-authored-by: vipul0425 <vipulgupta003@git.wordpress.org>
Co-authored-by: DAreRodz <darerodz@git.wordpress.org>

mikejolley committed:
Allow Dataform Select Box to have a custom empty option (#70894)

* Only define "select item" if blank value is not provided

* Update storybook

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

adamsilverstein committed:
merge trunk

adamsilverstein committed:
remove unused defaults, fix errors

t-hamano committed:
Accordion Blocks: Make block icons internal (#71229)

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: mikachan <mikachan@git.wordpress.org>

t-hamano committed:
Experimental: Enable Command Palette in admin dashboard (#71030)

* Experimental: Enable Command Palette in admin dashboard

* Update lib/experimental/commands.php

Co-authored-by: Shail Mehta <shailmehta25@gmail.com>

* Remove experimental setting

* Update docs

---------

Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: youknowriad <youknowriad@git.wordpress.org>
Co-authored-by: shail-mehta <shailu25@git.wordpress.org>
Co-authored-by: mtias <matveb@git.wordpress.org>

ciampo committed:
DateCalendar, DateRangeCalendar: use px instead of rem units, better theming support (#71248)

* DateCalendar, DateRangeCalendar: switch from rem to px for some internal sizes

* CHANGELOG

* Use variables when possible

* Use theme-aware variables

mikejolley committed:
DataForm: panel layout can open as `dropdown` or `modal` (#71212)

Co-authored-by: mikejolley <mikejolley@git.wordpress.org>
Co-authored-by: jameskoster <jameskoster@git.wordpress.org>
Co-authored-by: oandregal <oandregal@git.wordpress.org>

ciampo committed:
MenuItem: make accessible when disabled (#71251)

* MenuItem: make accessible when disabled

* CHANGELOG

* Update tests to account for different way to disable menu item

karthikeya-io committed:
Editor: Fix drag and drop to the top of the page (#71241)

* Fix: Guard drag-and-drop against missing block DOM nodes

* Fix: Ensure Post Comments Link always renders block wrapper

----

Co-authored-by: karthikeya-io <karthikeya01@git.wordpress.org>
Co-authored-by: talldan <talldanwp@git.wordpress.org>
Co-authored-by: tellthemachines <isabel_brison@git.wordpress.org>

im3dabasia committed:
Video: Enable autoplay for video on `iOS` devices (#69380)

Co-authored-by: im3dabasia <im3dabasia1@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: fluiddot <carlosgprim@git.wordpress.org>
Co-authored-by: gturpin-dev <guillaumeturpin@git.wordpress.org>

mrleemon committed:
Block Comments: Remove period in comment UI text (#71234)

* Fix typo in comments.js
* Remove period from `more replies..`


Co-authored-by: mrleemon <leemon@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>
Co-authored-by: akasunil <sunil25393@git.wordpress.org>
Co-authored-by: ramonjd <ramonopoly@git.wordpress.org>

mikachan committed:
Rename Accordions block to Accordion (#71222)

* Rename Accordions to Accordion

* Rename accordion-item

* Update block descriptions

* Tweak block descriptions

Co-authored-by: mikachan <mikachan@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: alexandrebuffet <alexandrebuffet@git.wordpress.org>
Co-authored-by: hanneslsm <hanneslsm@git.wordpress.org>
Co-authored-by: jasmussen <joen@git.wordpress.org>

oandregal committed:
DataForm: update stories (#71268)

Co-authored-by: oandregal <oandregal@git.wordpress.org>

Mamaduka committed:
Social: Include developer blog tutorial in README (#71266)


Co-authored-by: Mamaduka <mamaduka@git.wordpress.org>
Co-authored-by: bph <bph@git.wordpress.org>

USERSATOSHI committed:
Move Style above Navigation in Command Pallete (#71265)

Co-authored-by: USERSATOSHI <tusharbharti@git.wordpress.org>
Co-authored-by: karthick-murugan <karthickmurugan@git.wordpress.org>
Co-authored-by: t-hamano <wildworks@git.wordpress.org>
Co-authored-by: jordesign <jordesign@git.wordpress.org>

priethor committed:
Fix DataViews primary actions horizontal layout consistency (#71274)

* Fix DataViews actions horizontal layout consistency

* Update changelog with this fix

gutenbergplugin committed:
Bump plugin version to 21.5.0-rc.1

gutenbergplugin committed:
Update changelog files

gutenbergplugin committed:
chore(release): publish

 - @wordpress/a11y@4.29.0
 - @wordpress/annotations@3.29.0
 - @wordpress/api-fetch@7.29.0
 - @wordpress/autop@4.29.0
 - @wordpress/babel-plugin-import-jsx-pragma@5.29.0
 - @wordpress/babel-plugin-makepot@6.29.0
 - @wordpress/babel-preset-default@8.29.0
 - @wordpress/base-styles@6.5.0
 - @wordpress/blob@4.29.0
 - @wordpress/block-directory@5.29.0
 - @wordpress/block-editor@15.2.0
 - @wordpress/block-library@9.29.0
 - @wordpress/block-serialization-default-parser@5.29.0
 - @wordpress/block-serialization-spec-parser@5.29.0
 - @wordpress/blocks@15.2.0
 - @wordpress/browserslist-config@6.29.0
 - @wordpress/commands@1.29.0
 - @wordpress/components@30.2.0
 - @wordpress/compose@7.29.0
 - @wordpress/core-commands@1.29.0
 - @wordpress/core-data@7.29.0
 - @wordpress/create-block@4.72.0
 - @wordpress/create-block-interactive-template@2.29.0
 - @wordpress/create-block-tutorial-template@4.29.0
 - @wordpress/customize-widgets@5.29.0
 - @wordpress/data@10.29.0
 - @wordpress/data-controls@4.29.0
 - @wordpress/dataviews@7.0.0
 - @wordpress/date@5.29.0
 - @wordpress/dependency-extraction-webpack-plugin@6.29.0
 - @wordpress/deprecated@4.29.0
 - @wordpress/docgen@2.29.0
 - @wordpress/dom@4.29.0
 - @wordpress/dom-ready@4.29.0
 - @wordpress/e2e-test-utils@11.29.0
 - @wordpress/e2e-test-utils-playwright@1.29.0
 - @wordpress/e2e-tests@8.29.0
 - @wordpress/edit-post@8.29.0
 - @wordpress/edit-site@6.29.0
 - @wordpress/edit-widgets@6.29.0
 - @wordpress/editor@14.29.0
 - @wordpress/element@6.29.0
 - @wordpress/env@10.29.0
 - @wordpress/escape-html@3.29.0
 - @wordpress/eslint-plugin@22.15.0
 - @wordpress/fields@0.21.0
 - @wordpress/format-library@5.29.0
 - @wordpress/hooks@4.29.0
 - @wordpress/html-entities@4.29.0
 - @wordpress/i18n@6.2.0
 - @wordpress/icons@10.29.0
 - @wordpress/interactivity@6.29.0
 - @wordpress/interactivity-router@2.29.0
 - @wordpress/interface@9.14.0
 - @wordpress/is-shallow-equal@5.29.0
 - @wordpress/jest-console@8.29.0
 - @wordpress/jest-preset-default@12.29.0
 - @wordpress/jest-puppeteer-axe@7.29.0
 - @wordpress/keyboard-shortcuts@5.29.0
 - @wordpress/keycodes@4.29.0
 - @wordpress/lazy-import@2.29.0
 - @wordpress/list-reusable-blocks@5.29.0
 - @wordpress/media-utils@5.29.0
 - @wordpress/notices@5.29.0
 - @wordpress/npm-package-json-lint-config@5.29.0
 - @wordpress/nux@9.29.0
 - @wordpress/patterns@2.29.0
 - @wordpress/plugins@7.29.0
 - @wordpress/postcss-plugins-preset@5.29.0
 - @wordpress/postcss-themes@6.29.0
 - @wordpress/preferences@4.29.0
 - @wordpress/preferences-persistence@2.29.0
 - @wordpress/prettier-config@4.29.0
 - @wordpress/primitives@4.29.0
 - @wordpress/priority-queue@3.29.0
 - @wordpress/private-apis@1.29.0
 - @wordpress/project-management-automation@2.29.0
 - @wordpress/react-i18n@4.29.0
 - @wordpress/readable-js-assets-webpack-plugin@3.29.0
 - @wordpress/redux-routine@5.29.0
 - @wordpress/reusable-blocks@5.29.0
 - @wordpress/rich-text@7.29.0
 - @wordpress/router@1.29.0
 - @wordpress/scripts@30.22.0
 - @wordpress/server-side-render@6.5.0
 - @wordpress/shortcode@4.29.0
 - @wordpress/style-engine@2.29.0
 - @wordpress/stylelint-config@23.21.0
 - @wordpress/sync@1.29.0
 - @wordpress/token-list@3.29.0
 - @wordpress/undo-manager@1.29.0
 - @wordpress/upload-media@0.14.0
 - @wordpress/url@4.29.0
 - @wordpress/viewport@6.29.0
 - @wordpress/warning@3.29.0
 - @wordpress/widgets@4.29.0
 - @wordpress/wordcount@4.29.0

gutenbergplugin committed:
Update Changelog for 21.5.0-rc.1

adamsilverstein committed:
Merge branch 'trunk' into add/client-side-media-uploads

adamsilverstein committed:
Queueitem is an interface
